modules/index/src/test/java/org/apache/ignite/internal/index/IndexBuildControllerTest.java

```diff
@@ -0,0 +1,280 @@
+            MvPartitionStorage mvPartitionStorage = mock(MvPartitionStorage.class);
+            IndexStorage indexStorage = mock(IndexStorage.class);
+
+            when(mvTableStorage.getMvPartition(anyInt())).thenReturn(mvPartitionStorage);
+            when(mvTableStorage.getIndex(anyInt(), anyInt())).thenReturn(indexStorage);
+
+            return completedFuture(mvTableStorage);
+        });
+
+        clusterService = mock(ClusterService.class, invocation -> mock(TopologyService.class, invocation1 -> localNode));
+
+        catalogManager = CatalogTestUtils.createTestCatalogManager(NODE_NAME, clock);
+        catalogManager.start();
+
+        TableTestUtils.createTable(
+                catalogManager,
+                DEFAULT_SCHEMA_NAME,
+                DEFAULT_ZONE_NAME,
+                TABLE_NAME,
+                List.of(ColumnParams.builder().name(COLUMN_NAME).type(INT32).build()),
+                List.of(COLUMN_NAME)
+        );
+    }
+
+    @AfterEach
+    void tearDown() throws Exception {
+        IgniteUtils.stopAll(catalogManager);
+    }
+
+    @Test
+    void testStartBuildIndexesOnIndexCreate() {
+        newIndexBuildController();
+
+        setPrimaryReplicaForOneSecond(10, NODE_NAME, clock.now());
+
+        clearInvocations(indexBuilder);
+
+        createIndex(INDEX_NAME);
+
+        verify(indexBuilder).startBuildIndex(eq(tableId()), eq(10), eq(indexId(INDEX_NAME)), any(), any(), eq(localNode));
+    }
+
+    @Test
+    void testStartBuildIndexesOnPrimaryReplicaElected() {
+        newIndexBuildController();
+
+        createIndex(INDEX_NAME);
+
+        setPrimaryReplicaForOneSecond(10, NODE_NAME, clock.now());
+
+        verify(indexBuilder).startBuildIndex(eq(tableId()), eq(10), eq(indexId(INDEX_NAME)), any(), any(), eq(localNode));
+        verify(indexBuilder).startBuildIndex(eq(tableId()), eq(10), eq(indexId(TABLE_NAME + "_PK")), any(), any(), eq(localNode));
+    }
+
+    @Test
+    void testStopBuildIndexesOnIndexDrop() {
+        newIndexBuildController();
+
+        createIndex(INDEX_NAME);
+
+        int indexId = indexId(INDEX_NAME);
+
+        dropIndex(INDEX_NAME);
+
+        verify(indexBuilder).stopBuildingIndexes(indexId);
+    }
+
+    @Test
+    void testStopBuildIndexesOnChangePrimaryReplica() {
+        newIndexBuildController();
+
+        setPrimaryReplicaForOneSecond(10, NODE_NAME, clock.now());
+        setPrimaryReplicaForOneSecond(10, NODE_NAME + "_other", clock.now());
+
+        verify(indexBuilder).stopBuildingIndexes(tableId(), 10);
+    }
+
+    private void createIndex(String indexName) {
+        createHashIndex(catalogManager, DEFAULT_SCHEMA_NAME, TABLE_NAME, indexName, List.of(COLUMN_NAME), false);
+    }
+
+    private void dropIndex(String indexName) {
+        TableTestUtils.dropIndex(catalogManager, DEFAULT_SCHEMA_NAME, indexName);
+    }
+
+    private void setPrimaryReplicaForOneSecond(int partitionId, String leaseholder, HybridTimestamp startTime) {
```

Comment:
It's not clear what you mean by "for one second". Can you please elaborate in the comment?

Reply:
try fix it